app/jobs/pdf_generator.rb

```diff
@@ -0,0 +1,118 @@
+require 'json'
+require 'net/http'
+require 'rmagick'
+
+class PdfGenerator
+    @queue = :ingests
+
+    def self.perform(pid)
+      manifest_url = self.config['vufind_url'].to_s + "/Item/" + pid + "/Manifest"
+      json = self.do_get manifest_url
+      parsed_json = JSON.parse json
+      if (self.has_pdf_already parsed_json)
+        return
+      else
+        large_jpegs = parsed_json["sequences"][0]["canvases"].map do |current|
+          current["images"][0]["resource"]["@id"]
+        end
+        pdf = self.generate_pdf(large_jpegs)
+        self.add_pdf_to_pid(pdf,pid)
+      end
+    end
+
+    def self.has_pdf_already(parsed_json)
+      rendering_format = parsed_json["sequences"][0]["canvases"][0]["rendering"].map do |current|
+        current["format"]
+      end
+      if (rendering_format == "application/pdf")
```

Comment:
I don't think this is going to work; `rendering_format` is going to be an array at this point, so comparing it to the string `"application/pdf"` will always resolve to false. I think what you actually want to do is use the array's `include?` method... I think you can add the whole `if ... end` block with: `rendering_format.include? "application/pdf"`. If that's the last line of your function, the boolean coming back from `include?` will implicitly become the function's overall return value.

I'd definitely test this to be sure -- i.e. run this job twice on the same PID and make sure it doesn't add a second PDF -- but I think it should work.